lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -325,6 +359,23 @@ static void eraseControlWithGroupAndConditional(OpTy op,
     rewriter.eraseOp(cond.getDefiningOp());
 }
 
+/// A helper function to check whether the conditional needs to be erased
+/// to maintain a valid state of a Calyx program. If these
+/// have no more uses, they will be erased.
+template <typename OpTy>
+static void eraseControlWithConditional(OpTy op, PatternRewriter &rewriter) {
+  static_assert(std::is_same<OpTy, StaticIfOp>(),
```

Comment:
I'm surprised this isn't either std::is_same_v or suffixed by ::value()

Reply:
I'm pretty sure is_same will automatically cast to a bool without needing to use is_same_v, but let me know if I'm wrong.

Reply:
https://en.cppreference.com/w/cpp/types/is_same Since C++14, you're right!